src/test/Microsoft.NET.HostModel.Tests/Microsoft.NET.HostModel.AppHost.Tests/AppHostUpdateTests.cs

```diff
@@ -206,6 +242,69 @@ private string PrepareAppHostMockFile(TestDirectory testDirectory, Action<byte[]
             0, 112, 2, 0, 0, 4, 0, 0, 0, 0, 0, 0, 3, 0, 96, 193, 0, 0, 24,
             0, 0, 0, 0, 0, 0, 16, 0, 0, 0, 0 };
 
+        [DllImport("libc", SetLastError = true)]
+        private static extern int chmod(string pathname, int mode);
+
+        private static int GetFilePermissionOctal(string path)
+        {
+            var modeValue = CoreFxFileStatusProvider.GetFileMode(path);
+
+            // st_mode is typically a 16-bits value, high 4 bits are filetype and low 12
+            // bits are permission. we will clear first 20 bits (a byte and a nibble) with
+            // the following mask:
+            modeValue &= 0x1ff;
+
+            modeValue
+                .Should()
+                .BeInRange(0, 511);
+
+            // convert to octal number
+            int octal = (modeValue / 64) * 100;
+            modeValue %= 64;
+            octal += (modeValue / 8) * 10;
+            octal += modeValue % 8;
+            return octal;
```

Comment:
It seems more natural to round-trip this using `Convert.ToString(octal, 8)` (mirroring `Convert.ToInt32("444", 8)` above) and return a `string`--it's unclear to me from the method name that it's returning a base-10 reinterpretation of a base-8 number.

(Alternative: return the value as-is and use `Convert.ToInt32("444", 8)` in the test cases.)

Reply:
I have adapted to your alternative version. 👍